### PR TITLE
update doc for h3 Connection::send_response

### PR DESCRIPTION
**Pull request by keepsimple1**
This is to address issue #530 . Please review. Thanks.

### CODE REVIEW

**Comment by LPardue** [comment]
Generally looks good, thanks.

The CI is having a bit of trouble right now but I think it would fail due to rustfmt triggering errors due to line length.

I've made a suggestion of some slight tweaks in https://github.com/cloudflare/quiche/commit/1c0510617025cd4ed0d310fd6c5b4f8561273926. It addresses the above problem and also adds the additional comments to the send_request() method that behaves similarly.

**Comment by keepsimple1** [comment]
Addressed review comments and rebased.

File: src/h3/mod.rs [hunk 1]
```diff
@@ -588,8 +588,10 @@ impl Connection {
 
     /// Sends an HTTP/3 request.
     ///
-    /// The request is encoded from the provided list of headers, and sent on
-    /// a newly allocated stream.
+    /// The request is encoded from the provided list of headers without a
+    /// body, and sent on a newly allocated stream. To include a body,
+    /// set `fin` as `false` and subsequently call [`send_body`] with the
+    /// same connection and the stream_id returned from this method.
```
Comment (LPardue) [inline src/h3/mod.rs:594]:
```suggestion
    /// same `conn` and the `stream_id` returned from this method.
```

File: src/h3/mod.rs [hunk 2]
```diff
@@ -627,11 +630,16 @@ impl Connection {
 
     /// Sends an HTTP/3 response on the specified stream.
     ///
+    /// This method sends the provided `headers` without a body. To include a
+    /// body, set `fin` as `false` and subsequently call [`send_body`] with the
+    /// same connection and stream_id.
```
Comment (LPardue) [inline src/h3/mod.rs:635]:
```suggestion
    /// same `conn` and `stream_id`.
```

File: src/h3/mod.rs [hunk 3]
```diff
@@ -588,8 +588,10 @@ impl Connection {
 
     /// Sends an HTTP/3 request.
     ///
-    /// The request is encoded from the provided list of headers, and sent on
-    /// a newly allocated stream.
+    /// The request is encoded from the provided list of headers without a
+    /// body, and sent on a newly allocated stream. To include a body,
+    /// set `fin` as `false` and subsequently call [`send_body`] with the
```
Comment (LPardue) [inline src/h3/mod.rs:593]:
```suggestion
    /// set `fin` as `false` and subsequently call [`send_body()`] with the
```

File: src/h3/mod.rs [hunk 4]
```diff
@@ -627,11 +630,16 @@ impl Connection {
 
     /// Sends an HTTP/3 response on the specified stream.
     ///
+    /// This method sends the provided `headers` without a body. To include a
+    /// body, set `fin` as `false` and subsequently call [`send_body`] with the
```
Comment (LPardue) [inline src/h3/mod.rs:634]:
```suggestion
    /// body, set `fin` as `false` and subsequently call [`send_body()`] with the
```

File: src/h3/mod.rs [hunk 5]
```diff
@@ -627,11 +630,16 @@ impl Connection {
...
+    /// same connection and stream_id.
+    ///
     /// The [`StreamBlocked`] error is returned when the underlying QUIC stream
     /// doesn't have enough capacity for the operation to complete. When this
     /// happens the application should retry the operation once the stream is
     /// reported as writable again.
     ///
+    /// [`send_body`]: struct.Connection.html#method.send_body
```
Comment (LPardue) [inline src/h3/mod.rs:642]:
```suggestion
    /// [`send_body()`]: struct.Connection.html#method.send_body
```

**Comment by keepsimple1** [comment]
Travis CI failed with one error:
```
$ rustup component add rustfmt
error: component 'rustfmt' for target 'x86_64-unknown-linux-gnu' is unavailable for download for channel nightly
Sometimes not all components are available in any given nightly.
The command "rustup component add rustfmt" failed and exited with 1 during .
```

Is it a reliable test? How can I re-run the Travis CI?

**Comment by LPardue** [comment]
Its mostly reliable. All this test does is run the code linter, rustfmt, over the code. From time to time the package gets broken.

In this case you're just adding some comments, so as long as the column width is within bounds we should be fine to go ahead and merge.

**Comment by junhochoi** [comment]
> Travis CI failed with one error:

In CI we use nightly `rustfmt` but it's broken last few days : https://rust-lang.github.io/rustup-components-history/
That's why all recent PR show a broken status. Usually it will be resolved in a few days.
If all other build (stable and other platforms) is ok you are fine.

**Comment by keepsimple1** [comment]
I don't have permissions to merge the PR.  If you (reviewers) think it's good, could you please help merge it? 

And I have a question: why not to use the stable `rustfmt` instead of the nightly?

**Comment by ghedo** [comment]
Note that the nightly build for this PR failed because the code needs to be reformatted, see https://travis-ci.com/github/cloudflare/quiche/jobs/338131105#L1757

**Comment by keepsimple1** [comment]
I've updated the diff to fix the `rustfmt` error. Please review again. Thanks.

**Comment by ghedo** [comment]
Merged, thanks @keepsimple1!

**Pull request merged**